packages/rate-limit/rate-limit.js

```diff
@@ -133,7 +143,10 @@ RateLimiter.prototype.check = function (input) {
   var reply = {
     allowed: true,
     timeToReset: 0,
-    numInvocationsLeft: Infinity
+    numInvocationsLeft: Infinity,
+    userId: input.userId,
+    clientAddress: input.clientAddress,
+    connectionId: input.connectionId,
```

Comment:
I exclude `type` and `name` because I could not think of any useful use-cases for them. If it is still useful to have them, I will include them.